homeassistant/components/miele/sensor.py

```diff
@@ -0,0 +1,220 @@
+from .coordinator import MieleConfigEntry, MieleDataUpdateCoordinator
+from .entity import MieleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MieleSensorDescription(SensorEntityDescription):
+    """Class describing Miele sensor entities."""
+
+    data_tag: str
+    convert: Callable[[Any], Any] | None = None
+    zone: int | None = None
+
+
+@dataclass
+class MieleSensorDefinition:
+    """Class for defining sensor entities."""
+
+    types: tuple[MieleAppliance, ...]
+    description: MieleSensorDescription
+
+
+SENSOR_TYPES: Final[tuple[MieleSensorDefinition, ...]] = (
+    MieleSensorDefinition(
+        types=(
+            MieleAppliance.WASHING_MACHINE,
+            MieleAppliance.WASHING_MACHINE_SEMI_PROFESSIONAL,
+            MieleAppliance.TUMBLE_DRYER,
+            MieleAppliance.TUMBLE_DRYER_SEMI_PROFESSIONAL,
+            MieleAppliance.DISHWASHER,
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.HOB_HIGHLIGHT,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.COFFEE_SYSTEM,
+            MieleAppliance.HOOD,
+            MieleAppliance.FRIDGE,
+            MieleAppliance.FREEZER,
+            MieleAppliance.FRIDGE_FREEZER,
+            MieleAppliance.ROBOT_VACUUM_CLEANER,
+            MieleAppliance.WASHER_DRYER,
+            MieleAppliance.DISH_WARMER,
+            MieleAppliance.HOB_INDUCTION,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.WINE_CABINET,
+            MieleAppliance.WINE_CONDITIONING_UNIT,
+            MieleAppliance.WINE_STORAGE_CONDITIONING_UNIT,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.DIALOG_OVEN,
+            MieleAppliance.WINE_CABINET_FREEZER,
+            MieleAppliance.STEAM_OVEN_MK2,
+            MieleAppliance.HOB_INDUCT_EXTR,
+        ),
+        description=MieleSensorDescription(
+            key="state_status",
+            translation_key="status",
+            data_tag="state_status",
+            convert=lambda x: STATE_STATUS_TAGS.get(x, x),
+        ),
+    ),
+    MieleSensorDefinition(
+        types=(
+            MieleAppliance.TUMBLE_DRYER_SEMI_PROFESSIONAL,
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.DISH_WARMER,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.FRIDGE,
+            MieleAppliance.FREEZER,
+            MieleAppliance.FRIDGE_FREEZER,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.WINE_CABINET,
+            MieleAppliance.WINE_CONDITIONING_UNIT,
+            MieleAppliance.WINE_STORAGE_CONDITIONING_UNIT,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.DIALOG_OVEN,
+            MieleAppliance.WINE_CABINET_FREEZER,
+            MieleAppliance.STEAM_OVEN_MK2,
+        ),
```

Comment:
If there will be more of these lists of appliances, maybe consider looking at how lg_thinq does it. (I don't think there's a right way to do it and I think both ways have pros and cons, but it depends on how many there are and how many varieties)

Reply:
There is an A4 sheet from Miele with a capabilities matrix. Partly incorrect, by the way. I will look around.

Reply:
Looked around and I don't think the alternate solutions are significantly better.